backend-rust/src/indexer.rs

```diff
@@ -1989,18 +1989,27 @@ impl PreparedAccountDelegationEvent {
                     )
                     .execute(tx.as_mut())
                     .await?
-                    .ensure_affected_rows_in_range(bakers_expected_affected_range)
+                    .ensure_affected_rows_in_range(bakers_expected_affected_range.clone())
                     .context("Failed update pool stake adding delegator")?;
                 }
                 // Set the new target on the delegator.
+                // Prior to Protocol version 7, removing a baker was not immediate, but after
+                // some cooldown period, allowing delegators to still target the pool after
+                // removal. Since we remove the baker immediate even for older blocks there
+                // might not be a baker to target, so we check for existence as part of the
+                // query.
                 sqlx::query!(
-                    "UPDATE accounts SET delegated_target_baker_id = $1 WHERE index = $2",
+                    "UPDATE accounts
```

Comment:
This means the `old_delegation_target` stays for the delgators? Would that not also cause long-term problems?

Reply:
The old ones are cleaned up in migration 5 and the constraint ensures this stays true, which is also why it fails at the moment.
We are just missing handling of delegators targeting bakers after they are set to be removed.